services/src/main/java/org/keycloak/utils/SecureContextResolver.java

```diff
@@ -79,16 +78,15 @@ public static boolean isLocalAddress(String address) {
         if (address == null) {
             return false;
         }
-        // The host matches a CIDR notation of ::1/128
-        if (address.startsWith("[")) {
-            return LOCALHOST_IPV6.matcher(address).matches();
-        }
 
-        // The host matches a CIDR notation of 127.0.0.0/8
-        if (LOCALHOST_IPV4.matcher(address).matches()) {
-            return true;
+        if (NetUtil.isValidIpV4Address(address) || NetUtil.isValidIpV6Address(address)) {
+            try {
+                return InetAddress.getByName(address).isLoopbackAddress();
```

Comment:
`InetAddress#getByName` is doing a DNS request, no? At least when it's called from [`isSecureContext`](https://github.com/keycloak/keycloak/blob/63a87155478b0df8ce2a9e0eb45715d2373b0090/services/src/main/java/org/keycloak/utils/SecureContextResolver.java#L61). Not sure we should do it as it might be partially a regression of #35216, no?

Reply:
> InetAddress#getByName is doing a DNS request, no?

Only for non-IP values.

Reply:
Yes, and that was the issue with #35216 as well, right?

Reply:
Right, the `isValidIp...` method. I guess my brain is still booting up in the 2025, sorry. :)